docs/readthedocs/source/doc/DLlib/Overview/getting-started.md

```diff
@@ -0,0 +1,245 @@
+The IDE will automatically download it from maven and you are able to run your application.
+
+
+## 2. Code initialization
+```NNContext``` is the main entry for provisioning the dllib program on the underlying cluster (such as K8s or Hadoop cluster), or just on a single laptop.
+
+We usually do some initialization at first
+```
+import com.intel.analytics.bigdl.dllib.NNContext
+
+val sc = NNContext.initNNContext("dllib_demo")
+```
+For more information about ```NNContext```, please refer to [NNContext](https://bigdl.readthedocs.io/en/latest/doc/DLlib/Overview/dllib.html#nn-context)
+
+## 3. Distributed Data Processing
+
+#### Using Spark Dataframe APIs
+DLlib supports Spark Dataframes as the input to the distributed training, and as
+the input/output of the distributed inference. Consequently, the user can easily
+process large-scale dataset using Apache Spark, and directly apply AI models on
+the distributed (and possibly in-memory) Dataframes without data conversion or serialization
+
+We used [Pima Indians onset of diabetes](https://raw.githubusercontent.com/jbrownlee/Datasets/master/pima-indians-diabetes.data.csv) as dataset for the demo. It's a standard machine learning dataset from the UCI Machine Learning repository. It describes patient medical record data for Pima Indians and whether they had an onset of diabetes within five years.
+The dataset can be download with:
+```
+wget https://raw.githubusercontent.com/jbrownlee/Datasets/master/pima-indians-diabetes.data.csv
+```
+
+We create Spark session so we can use Spark API to load and process the data
+```
+import org.apache.spark.sql.SQLContext
+val spark = new SQLContext(sc)
+```
+
+Load the data into Spark DataFrame
+```
+val path = "pima-indians-diabetes.data.csv"
+val df = spark.read.options(Map("inferSchema"->"true","delimiter"->",")).csv(path)
+      .toDF("num_times_pregrant", "plasma_glucose", "blood_pressure", "skin_fold_thickness", "2-hour_insulin", "body_mass_index", "diabetes_pedigree_function", "age", "class")
+```
+
+Process the DataFrame to create the label and features. Then split it into traing part and validation part
+
+```
+val assembler = new VectorAssembler()
+  .setInputCols(Array("num_times_pregrant", "plasma_glucose", "blood_pressure", "skin_fold_thickness", "2-hour_insulin", "body_mass_index", "diabetes_pedigree_function", "age"))
+  .setOutputCol("features")
+val assembleredDF = assembler.transform(df)
+val df2 = assembleredDF.withColumn("label",col("class").cast(DoubleType) + lit(1))
+val Array(trainDF, valDF) = df2.randomSplit(Array(0.8, 0.2))
+```
+
+Now we have got the data which is ready to train. Next we will build a deep learning model using DLLib Keras API
+
+## 4. Model Definition
+
+#### Using Keras-like APIs
+
+To define a model, you can use the [Keras Style API](https://bigdl.readthedocs.io/en/latest/doc/DLlib/Overview/keras-api.html).
+```
+import com.intel.analytics.bigdl.dllib.keras.layers._
+val x1 = Input[Float](Shape(8))
```

Comment:
The user needs not to use `[Float]` in the program; instead, I think the user can just `import com.intel.analytics.bigdl.numeric.NumericFloat`, and then the type can be inferred.